packages/run-protocol/src/vaultFactory/liquidateIncrementally.js

```diff
@@ -242,16 +252,28 @@ const start = async zcf => {
     trace('offerResult', { amounts });
   }
 
-  const debtorHook = async (debtorSeat, { debt: originalDebt }) => {
-    trace('LIQ', originalDebt);
+  /**
+   * @param {ZCFSeat} debtorSeat
+   * @param {object} options
+   * @param {Amount<'nat'>} options.debt Debt before penalties
+   * @param {Ratio} options.penaltyRate
+   */
+  const handleLiquidateOffer = async (
+    debtorSeat,
+    { debt: originalDebt, penaltyRate },
+  ) => {
     assertProposalShape(debtorSeat, {
       give: { In: null },
     });
     assert(
       originalDebt.brand === debtBrand,
       X`Cannot liquidate to ${originalDebt.brand}`,
     );
-    for await (const t of processTranches(debtorSeat, originalDebt)) {
+    const penalty = ceilMultiplyBy(originalDebt, penaltyRate);
+    const debtWithPenalty = AmountMath.add(originalDebt, penalty);
+    trace('LIQ', { originalDebt, debtWithPenalty });
+
+    for await (const t of processTranches(debtorSeat, debtWithPenalty)) {
```

Comment:
Good catch!  Did changing to `debtWithPenalty` break any tests?

I don't see any related changes, so if not, would you mind adding one that would have detected the difference?

Reply:
As you mentioned, the behavior doesn't change here, the debt calculation just moved to the callee rather than the caller.